### ISSUE TITLE
[idea] virtual files (or tabs) inside cartridges.. for organize libs, classes, etc

### ISSUE THREAD

**Issue by atesin** [nesbox/TIC-80#2414]
that is... often when our code grows it got more difficult to navigate, and coding turn less efficient tangled and consuming

there are some code snippets, functions, libs, classes, etc, that would be much better organized in separate files (or TABS)... often in more "serious" programs these elements are separated in files/packages and invoked with `include` or `require`

i read tic80 also supports `require` function to load external lua files (and other languages?), but these must also be distributed along with cartridge in order to function correctly :(   ...  this idea comes to surpasses this limitation while keep code organized and easy at same time (**like in arduino ide**)

maybe inside tic80 the `require` function is not even needed, should be enough with dividing entire code area in cartridge storage/ram in "pages" with some kind of deliniter (like an unlikely custom comment like `-- tab1 foobarbazqux ` for example) that is read as a whole by interpreter, but editor separates in tabs

the only real care i see is the `import/export code file.lua` new format, could be `code1, code2, code3, etc...` new paramenters

**Comment by blueloveTH** [comment]
I agree with this. If tic-80 supports virtual files, `import` can be used in python, too!

**Comment by atesin** [comment]
i imagine the editor are watching for user puts a comment like `-- page: <name>` and then separate code in another tab, so the user just need to add or delete that comment to editor show code paged in tabs... like:

![imagen](https://github.com/nesbox/TIC-80/assets/8086106/2b801e15-84c7-48de-a963-536281cd1291)

then the user inserts a comment like

![imagen](https://github.com/nesbox/TIC-80/assets/8086106/c7f997c6-44ed-4e2a-8927-81ffb454a476)

and the editor create a tab and show the code from this comment below

![imagen](https://github.com/nesbox/TIC-80/assets/8086106/75a94cfc-61c9-4a22-b692-7cdd8b3c88fa)

hovering the "shapes" tab shows a list of all active tabs..... 

![imagen](https://github.com/nesbox/TIC-80/assets/8086106/2a1739ae-e760-499e-a5a4-fe2acabe3e16)

and to delete a tab (merge 2 consecutive tabs) is enough with delete the  `-- page: <name>` comment at the top

**Comment by TripleCubes** [comment]
Isnt this just bookmark with extra steps? I think the real solution to this is to make bookmarks saved between sessions

**Comment by runemoennike** [comment]
Just some oldschool input: in qbasic, each function/subroutine was separated and the editor showed only one at a time. A hotkey would bring up a window with the full list, sorted by module if other modules had been imported. I remember being shocked when I moved to other IDEs and realised this was not normal 😄
